src/app/pages/level-listing/level-listing.component.ts

```diff
@@ -28,40 +28,37 @@ export class LevelListingComponent implements OnInit {
     this.route.paramMap.subscribe((params: ParamMap) => {
       const apiRoute: string | null = params.get('route');
       if(apiRoute == null) return;
-
-      this.routeName = apiRoute;
-      this.processedRoute = this.processRoute(apiRoute);
-
+      this.apiRoute = apiRoute;
+      console.log(this.apiRoute)
       const pipe = this.apiClient.GetLevelListing(apiRoute, pageSize, 0)
         .pipe(catchError((error: HttpErrorResponse, caught) => {
           console.warn(error)
           if(error.status === 404) {
             this.router.navigate(["/404"]);
             return of();
           }
-
           return caught;
         }));
 
         pipe.subscribe(data => {
           this.levels = data.items;
           this.total = data.listInfo.totalItems;
         })
+
+      const categoryPipe = this.apiClient.GetLevelCategories().pipe(catchError((error: HttpErrorResponse, caught) => {
+        console.warn(error)
+        return caught;
```

Comment:
This is what we usually do, but I'm not sure if this is correct

```suggestion
        return of(undefined);
```

Reply:
This won't work, you have to somehow return the categories

Reply:
If there's an error, then you're not returning the categories anyways